### ISSUE TITLE
CLI debugger does not expose local scope until stepping past breakpoints

### ISSUE THREAD

**Issue by jeffomatic** [facebook/hhvm#8542]
### HHVM Version

```
$ hh_client --version
hh-a6226a116f4b94b47b893bbcf3a1a2b65649466d-4.0.4 Wed May 22 16:32:10 2019
```

### Operating System and Version

MacOS Mojave, 10.14.5.

### Standalone code, or other way to reproduce the problem

This is a very simple script that contains a breakpoint.

```
<?hh // strict

namespace Main;

function foo(int $bar): void {
	\hphpd_break();
	print "{$bar}\n";
}

<<__EntryPoint>>
function main(): void {
	foo(1);
}
```

### Actual result

Note that attempting to read `$bar` at the breakpoint yields an undefined variable error.

```
$ hhvm -a ./main.hh
Welcome to HipHop Debugger!
Type "help" or "?" for a complete list of commands.

Program ./main.hh loaded. Type '[r]un' or '[c]ontinue' to go.
hphpd> c
c
Break at Main\foo() on line 6 of /Users/jefflee/src/tmp/hhtest/main.hh
   5 function foo(int $bar): void {
   6* \hphpd_break();
   7  print "{$bar}\n";

hphpd> =$bar
=$bar

Notice: Undefined variable: bar
...(omitted)
hphpd> n
n
Break at Main\foo() on line 7 of /Users/jefflee/src/tmp/hhtest/main.hh
   6  \hphpd_break();
   7* print "{$bar}\n";
   8 }

hphpd> =$bar
=$bar
1
hphpd>
```

### Expected result

It would be nice if variables were readable without requiring a step past the breakpoint.

**Comment by fredemmott** [comment]
This is fixed in 4.12+; would it be useful to backport this change to 4.8, 4.10, and 4.11?

**Comment by jeffomatic** [comment]
Thanks for the update Fred. Happy to close this since I know it's coming down the line. A backport would be nice, but I don't even know how soon we'll be on 4.8 at this point. Speaking just for myself, I think this is more papercut than fire so it probably doesn't merit an extreme amount of effort.
